[2023-10-30T11:22:12.431+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: random_names.streaming_to_kafka manual__2023-10-30T11:22:06.472299+00:00 [queued]>
[2023-10-30T11:22:12.440+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: random_names.streaming_to_kafka manual__2023-10-30T11:22:06.472299+00:00 [queued]>
[2023-10-30T11:22:12.440+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 1
[2023-10-30T11:22:12.481+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): streaming_to_kafka> on 2023-10-30 11:22:06.472299+00:00
[2023-10-30T11:22:12.491+0000] {standard_task_runner.py:57} INFO - Started process 287 to run task
[2023-10-30T11:22:12.498+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'random_names', 'streaming_to_kafka', 'manual__2023-10-30T11:22:06.472299+00:00', '--job-id', '46', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpcq259i69']
[2023-10-30T11:22:12.502+0000] {standard_task_runner.py:85} INFO - Job 46: Subtask streaming_to_kafka
[2023-10-30T11:22:12.569+0000] {task_command.py:415} INFO - Running <TaskInstance: random_names.streaming_to_kafka manual__2023-10-30T11:22:06.472299+00:00 [running]> on host df75f57c0534
[2023-10-30T11:22:12.667+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='random_names' AIRFLOW_CTX_TASK_ID='streaming_to_kafka' AIRFLOW_CTX_EXECUTION_DATE='2023-10-30T11:22:06.472299+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-10-30T11:22:06.472299+00:00'
[2023-10-30T11:22:13.904+0000] {logging_mixin.py:151} INFO - {'full name': 'Mr Ezra Wright', 'gender': 'male', 'location': 'Symonds Street 7289', 'city': 'Christchurch', 'country': 'New Zealand', 'postcode': 84383, 'latitude': '54.8041', 'longtitude': '-60.3338', 'email': 'ezra.wright@example.com'}
[2023-10-30T11:22:13.911+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: connecting to localhost:29092 [('127.0.0.1', 29092) IPv4]
[2023-10-30T11:22:13.913+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2023-10-30T11:22:13.915+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]> returned error 111. Disconnecting.
[2023-10-30T11:22:13.915+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2023-10-30T11:22:13.960+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: connecting to localhost:29092 [('127.0.0.1', 29092) IPv4]
[2023-10-30T11:22:13.961+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]> returned error 111. Disconnecting.
[2023-10-30T11:22:13.961+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv4 ('127.0.0.1', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2023-10-30T11:22:14.010+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]>: connecting to localhost:29092 [('::1', 29092, 0, 0) IPv6]
[2023-10-30T11:22:14.011+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]> returned error 99. Disconnecting.
[2023-10-30T11:22:14.011+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=localhost:29092 <connecting> [IPv6 ('::1', 29092, 0, 0)]>: Closing connection. KafkaConnectionError: 99 EADDRNOTAVAIL
[2023-10-30T11:22:14.011+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/scripts/script.py", line 50, in start_streaming
    producer = get_kafka_producer()
  File "/opt/airflow/dags/scripts/script.py", line 31, in get_kafka_producer
    producer = KafkaProducer(
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/producer/kafka.py", line 381, in __init__
    client = KafkaClient(metrics=self._metrics, metric_group_prefix='producer',
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/client_async.py", line 244, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/airflow/.local/lib/python3.8/site-packages/kafka/client_async.py", line 900, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
[2023-10-30T11:22:14.122+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=random_names, task_id=streaming_to_kafka, execution_date=20231030T112206, start_date=20231030T112212, end_date=20231030T112214
[2023-10-30T11:22:14.183+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 46 for task streaming_to_kafka (NoBrokersAvailable; 287)
[2023-10-30T11:22:14.236+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-30T11:22:14.289+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
